sql/src/main/java/io/crate/jobs/JobExecutionContext.java

```diff
@@ -162,6 +163,15 @@ public void close() {
         }
     }
 
+    public void kill() {
+        Iterator<Map.Entry<Integer, ExecutionSubContext>> iterator = subContexts.entrySet().iterator();
```

Comment:
same here: imho we must prevent creating new contexts while kill, also while close imho